src/engine/controls/bpmcontrol.cpp

```diff
@@ -867,6 +867,9 @@ double BpmControl::getBeatMatchPosition(
     const double dOtherPosition = pOtherEngineBuffer->getExactPlayPos();
     const double dThisSampleRate = m_pBeats->getSampleRate();
     const double dThisRateRatio = m_pRateRatio->get();
+    if (dThisRateRatio == 0.0) {
```

Comment:
Please move this early return to the earliest point where a decision could be made and avoid calculating values that are not used. This is confusing and suggests that the preceding operations might have intended side effects which is fortunately not the case here.